reference/events.v1.yaml

```diff
@@ -9,14 +9,18 @@ info:
 
     The Event API is the mechanism, in conjunction with the Webhook API, that enables this event based notification mechanism to function reliably.  It enables subscribers to request events from the provider, and paginate through to the last place that it received a message.
 
+    The events are posted to the URL that the eventhook registers. The body of the post contains a list of new events to be processed.
+
     Notes:
       - Pagination is essential - using start and limit semantics.
+      - Oldest events first
       - Limits should be imposed on page size.
-      - Order should always be reverese chronological.
       - Filtering is available by matching specific event type.
-      - Providers should ensure that this API is cacheable or able to be retrieved at minimal cost, as it may be accessed a lot by consumers who are 'catching up'
+      - Filtering is available on event time, so only events after a date can be requested to catchup
+      
+    Providers should ensure that this API is cacheable or able to be retrieved at minimal cost, as it may be accessed a lot by consumers who are 'catching up'
```

Comment:
I see the point, but also makes me wonder how often one party who is catching up will retrieve the same events. And if there are multiple parties catching up, will they receive the same events? (except for the catalog/course, I would say no)

Reply:
I assume that who gets which event is handled by the implementer. Either make one queue and keep track which parties can see the event (and therefore have received it before), or make separate event queues per party.

A party catching up will not get the same events as each event is unique (timestand, objectid combination). The data in the event can be the same, so the receiving party can decide not to process events  based on the objectid and timestamp), but that partie should get all events first and then remove the ones on which the data is duplicate (and only keep the last). (this goes for sis and product data events, not for usage and progress)

The sending party can of course remove old events from the eventqueue if there is a new events for the same data. For me this is up to the implementors.